File: wisefy/src/androidTest/java/com/isupatches/wisefy/SearchForAccessPointTests.java
//package com.isupatches.wisefy;
//
//
//import android.net.wifi.ScanResult;
//import com.isupatches.wisefy.base.TestActivity;
//import com.isupatches.wisefy.callbacks.SearchForAccessPointCallbacks;
//import com.isupatches.wisefy.constants.WiseFyCodes;
//import org.junit.Test;
//import static com.isupatches.wisefy.base.TestUtils.TEST_SSID;
//import static junit.framework.Assert.assertEquals;
//import static junit.framework.Assert.fail;
//import static org.mockito.ArgumentMatchers.anyBoolean;
//import static org.mockito.ArgumentMatchers.anyInt;
//import static org.mockito.ArgumentMatchers.anyString;
//import static org.mockito.Mockito.mock;
//import static org.mockito.Mockito.timeout;
//import static org.mockito.Mockito.verify;
//import static org.mockito.Mockito.when;
//
//
//public class SearchForAccessPointTests extends BaseTestClass<TestActivity> {
//
//    public SearchForAccessPointTests() {
//        super(TestActivity.class);
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_false_nullSSIDParam() {
//        assertEquals(null, mWiseFy.searchForAccessPoint(null, 1000, false));
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_true_nullSSIDParam() {
//        assertEquals(null, mWiseFy.searchForAccessPoint(null, 1000, true));
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_false_missingPrerequisite() {
//        missingPrerequisite();
//        assertEquals(null, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false));
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_true_missingPrerequisite() {
//        missingPrerequisite();
//        assertEquals(null, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true));
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_false() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        assertEquals(null, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false));
//    }
//
//    @Test
//    public void noCallbacks_failure_filterDuplicates_true() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        assertEquals(null, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true));
//    }
//
//    @Test
//    public void noCallbacks_success_filterDuplicates_false() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        assertEquals(accessPoint, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false));
//    }
//
//    @Test
//    public void noCallbacks_success_filterDuplicates_true() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//        accessPoint.level = -35;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        assertEquals(accessPoint, mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true));
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false_nullSSIDParam() {
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(null, 1000, false, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).searchForAccessPointWiseFyFailure(WiseFyCodes.MISSING_PARAMETER);
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false_nullSSIDParam_nullCallback() {
//        try {
//            mWiseFy.searchForAccessPoint(null, 1000, false, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true_nullSSIDParam() {
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(null, 1000, true, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).searchForAccessPointWiseFyFailure(WiseFyCodes.MISSING_PARAMETER);
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true_nullSSIDParam_nullCallback() {
//        try {
//            mWiseFy.searchForAccessPoint(null, 1000, true, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false_missingPrerequisite() {
//        missingPrerequisite();
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).searchForAccessPointWiseFyFailure(WiseFyCodes.MISSING_PREREQUISITE);
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false_missingPrerequisite_nullCallback() {
//        missingPrerequisite();
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true_missingPrerequisite() {
//        missingPrerequisite();
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).searchForAccessPointWiseFyFailure(WiseFyCodes.MISSING_PREREQUISITE);
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true_missingPrerequisite_nullCallback() {
//        missingPrerequisite();
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).accessPointNotFound();
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_false_nullCallback() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).accessPointNotFound();
//    }
//
//
//    @Test
//    public void callbacks_failure_filterDuplicates_true_nullCallback() {
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(null);
//
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_success_filterDuplicates_false() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//        accessPoint.level = -35;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).accessPointFound(accessPoint);
//    }
//
//    @Test
//    public void callbacks_success_filterDuplicates_false_nullCallback() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//        accessPoint.level = -35;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, false, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//
//    @Test
//    public void callbacks_success_filterDuplicates_true() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//        accessPoint.level = -35;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        SearchForAccessPointCallbacks mockCallbacks = mock(SearchForAccessPointCallbacks.class);
//        mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, mockCallbacks);
//        verify(mockCallbacks, timeout(VERIFICATION_TIMEOUT)).accessPointFound(accessPoint);
//    }
//
//    @Test
//    public void callbacks_success_filterDuplicates_true_nullCallback() {
//        ScanResult accessPoint = mock(ScanResult.class);
//        accessPoint.SSID = TEST_SSID;
//        accessPoint.level = -35;
//
//        WiseFySearch mockWiseFySearch = mock(WiseFySearch.class);
//        mWiseFy.mWiseFySearch = mockWiseFySearch;
//        when(mockWiseFySearch.findAccessPointByRegex(anyString(), anyInt(), anyBoolean())).thenReturn(accessPoint);
//
//        try {
//            mWiseFy.searchForAccessPoint(TEST_SSID, 1000, true, null);
//        } catch (NullPointerException npe) {
//            fail();
//        }
//    }
//}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/SearchForAccessPointTests.kt
package com.isupatches.wisefy

import com.isupatches.wisefy.callbacks.SearchForAccessPointCallbacks
import com.isupatches.wisefy.constants.MISSING_PARAMETER
import com.isupatches.wisefy.internal.base.BaseInstrumentationTest

import org.junit.Assert.assertEquals
import org.junit.Test

import org.mockito.Mockito.mock
import org.mockito.Mockito.timeout
import org.mockito.Mockito.verify

/**
 * Used to test the functionality to search for a set of nearby access points.
 *
 * @author Patches
 * @since 3.0
 */
internal class SearchForAccessPointTests : BaseInstrumentationTest() {

    @Test fun sync_failure_prechecks_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        assertEquals(null, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false))
    }

    @Test fun sync_failure_prechecks_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        assertEquals(null, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true))
    }

    @Test fun sync_failure_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        assertEquals(null, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false))
    }

    @Test fun sync_failure_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        assertEquals(null, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true))
    }

    @Test fun sync_success_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        val accessPoint = mockWiseFySearchUtil.findAccessPointByRegex_success()
        assertEquals(accessPoint, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false))
    }

    @Test fun sync_success_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        val accessPoint = mockWiseFySearchUtil.findAccessPointByRegex_success()
        assertEquals(accessPoint, wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true))
    }

    @Test fun async_failure_prechecks_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).wisefyFailure(MISSING_PARAMETER)
    }

    @Test fun async_failure_prechecks_filterDuplicates_false_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, false)
    }

    @Test fun async_failure_prechecks_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).wisefyFailure(MISSING_PARAMETER)
    }

    @Test fun async_failure_prechecks_filterDuplicates_true_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_failure()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, true)
    }

    @Test fun async_failure_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).accessPointNotFound()
    }

    @Test fun async_failure_filterDuplicates_false_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, false)
    }

    @Test fun async_failure_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).accessPointNotFound()
    }

    @Test fun async_failure_filterDuplicates_true_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_null()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, true)
    }

    @Test fun async_success_filterDuplicates_false() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        val accessPoint = mockWiseFySearchUtil.findAccessPointByRegex_success()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, false, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).accessPointFound(accessPoint)
    }

    @Test fun async_success_filterDuplicates_false_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_success()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, false)
    }

    @Test fun async_success_filterDuplicates_true() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        val accessPoint = mockWiseFySearchUtil.findAccessPointByRegex_success()
        val mockCallbacks = mock(SearchForAccessPointCallbacks::class.java)
        wisefy.searchForAccessPoint(TEST_SSID, TEST_TIMEOUT, true, mockCallbacks)
        verify(mockCallbacks, timeout(VERIFICATION_SUCCESS_TIMEOUT)).accessPointFound(accessPoint)
    }

    @Test fun async_success_filterDuplicates_true_nullCallback() {
        mockWiseFyPrechecksUtil.searchForAccessPoint_success()
        mockWiseFySearchUtil.findAccessPointByRegex_success()
        nullCallbackUtil.callSearchForAccessPoint(TEST_SSID, true)
    }
}
